packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CreateFolderView/CreateFolderView.tsx

```diff
@@ -1,141 +1,91 @@
-  const [
-    {
-      data: { result },
-    },
-  ] = useAction('CREATE_FOLDER');
-
-  switch (result?.status) {
-    case 'COMPLETE':
-      return <Message variant="success">{RESULT_COMPLETE_MESSAGE}</Message>;
-    case 'FAILED':
-      return (
-        <Message variant="error">
-          {result.message ? result.message : RESULT_FAILED_MESSAGE}
-        </Message>
-      );
-    default:
-      return null;
-  }
-};
+  !!name?.length && !name.includes('/') && !name.includes('.');
 
-export const CreateFolderView = ({
+export function CreateFolderView({
   className,
-  onExit: onExitProps,
-}: CreateFolderViewProps): React.JSX.Element => {
-  const [{ location }, dipatchStoreAction] = useStore();
-  const { current, key } = location;
-
-  const { prefix } = current ?? {};
-  const hasInvalidPrefix = isUndefined(prefix);
-
-  const [{ isLoading, data }, handleCreateAction] = useAction('CREATE_FOLDER');
-  const { result } = data;
-
-  const [folderName, setFolderName] = React.useState('');
-  const [fieldValidationError, setFieldValidationError] = React.useState<
+  ...props
+}: CreateFolderViewProps): React.JSX.Element {
+  const {
+    getActionCompleteMessage,
+    getValidationMessage,
+    actionStartLabel,
+    folderNameLabel,
+    folderNamePlaceholder,
+  } = useDisplayText()['CreateFolderView'];
+
+  const {
+    folderName,
+    folderNameId,
+    isProcessing,
+    isProcessingComplete,
+    onActionStart,
+    onExit,
+    onFolderNameChange,
+    statusCounts,
+  } = useCreateFolderView(props);
+
+  const [validationMessage, setValidationMessage] = React.useState<
     string | undefined
   >();
 
-  const handleBlur = () => {
-    if (!isValidFolderName(folderName)) {
-      setFieldValidationError(FIELD_VALIDATION_MESSAGE);
-    }
-  };
-
-  const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
-    // validate on change if validation error is present
-    if (fieldValidationError && isValidFolderName(target.value)) {
-      setFieldValidationError(undefined);
-    }
-    setFolderName(target.value);
-  };
-
-  const handleCreateFolder = () => {
-    if (hasInvalidPrefix) return;
-    const folderPrefix = `${key}${folderName}/`;
-    handleCreateAction({ prefix: folderPrefix });
-  };
-
-  const handleClose = () => {
-    onExitProps?.(current);
-    dipatchStoreAction({ type: 'RESET_ACTION_TYPE' });
-    // reset hook state on exit, use empty string for prefix to keep TS happy
-    // @todo: this needs to be addressed
-    handleCreateAction({ prefix: '', options: { reset: true } });
+  const messageContent = isProcessingComplete
+    ? getActionCompleteMessage(statusCounts)
+    : undefined;
+
+  const contextValue: ControlsContext = {
+    data: {
+      folderNameId,
+      folderNameLabel,
+      folderNamePlaceholder,
+      folderNameValidationMessage: validationMessage,
+      actionStartLabel,
+      isActionStartDisabled:
+        !folderName.length ||
```

Comment:
super nit, could probably calculate `isActionStartDisabled` logic separately (similar to `messageContent` above) to keep `contextValue` concise